### PR TITLE
testing/gpg-remailer: new aport

### PR DESCRIPTION
**Pull request by Shizmob**
This PR adds `gpg-remailer` and its dependencies as new aports.

### CODE REVIEW

File: testing/bobcat/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Contributor: Shiz <hi@shiz.me>
+# Maintainer: Shiz <hi@shiz.me>
+pkgname=bobcat
+pkgver=4.07.00
+pkgrel=0
+pkgdesc="library of C++ classes and templates handling child processes, streams/sockets, shared memory, config files, etc."
```
Comment (jirutka) [inline testing/bobcat/APKBUILD]: Could you please make it a bit shorter, less than 100 chars? And start with a capital letter.

File: testing/bobcat/avoid-strerror_r.patch [hunk 2]
```diff
@@ -0,0 +1,20 @@
+--- a/exception/errnodescr.cc
```
Comment (jirutka) [inline testing/bobcat/avoid-strerror_r.patch]: Please add a description why is this patch needed.

File: testing/gpg-remailer/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Contributor: Shiz <hi@shiz.me>
+# Maintainer: Shiz <hi@shiz.me>
+pkgname=gpg-remailer
+pkgver=3.02.01
+pkgrel=0
+pkgdesc="decrypt messages and re-encrypt them to a well-defined group"
```
Comment (jirutka) [inline testing/gpg-remailer/APKBUILD]: Please start pkgdesc with a capital letter.

File: testing/gpg-remailer/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+url="https://fbb-git.github.io/gpg-remailer/"
+arch="all"
+license="GPL3+"
+makedepends="icmake bash yodl bobcat bobcat-dev"
```
Comment (jirutka) [inline testing/gpg-remailer/APKBUILD]: `foo-dev` package should typically depend on `foo`, so `bobcat` should not be needed here, only `bobcat-dev`.

File: testing/bobcat/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+url="https://fbb-git.github.io/bobcat/"
+arch="all"
+license="GPL2+"
+makedepends="icmake bash yodl"
+subpackages="$pkgname-doc $pkgname-dev"
+options="!check"
```
Comment (jirutka) [inline testing/bobcat/APKBUILD]: When disabling `check`, please add a short comment why, e.g. `options="!check"  # upstream does not provide tests`.

File: testing/gpg-remailer/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+subpackages="$pkgname-doc"
+source="gpg-remailer-$pkgver.tar.gz::https://github.com/fbb-git/gpg-remailer/archive/$pkgver.tar.gz
+	avoid-gnuisms.patch"
+builddir="$srcdir/gpg-remailer-$pkgver/gpg-remailer"
+options="!check"
```
Comment (jirutka) [inline testing/gpg-remailer/APKBUILD]: If upstream does not provide any tests and the project is an application (not a library), then it’s better to add at least very basic check that just calls the produced binary with `--version`. Just to check that the produced binary is not totally broken.

File: testing/icmake/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# Contributor: Shiz <hi@shiz.me>
+# Maintainer: Shiz <hi@shiz.me>
+pkgname=icmake
+pkgver=9.02.02
+pkgrel=0
+pkgdesc="a make utility using a C-like grammar"
```
Comment (jirutka) [inline testing/icmake/APKBUILD]: Please start pkgdesc with a capital letter.

File: testing/icmake/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+url="https://fbb-git.github.io/icmake/"
+arch="all"
+license="GPL3+"
+makedepends="bash"
+subpackages="$pkgname-doc"
+source="icmake-$pkgver.tar.gz::https://github.com/fbb-git/icmake/archive/$pkgver.tar.gz"
+builddir="$srcdir/icmake-$pkgver/icmake"
+options="!check"
```
Comment (jirutka) [inline testing/icmake/APKBUILD]: If upstream does not provide any tests and the project is an application (not a library), then it’s better to add at least very basic check that just calls the produced binary with `--version`. Just to check that the produced binary is not totally broken.

File: testing/icmake/APKBUILD [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+prepare() {
```
Comment (jirutka) [inline testing/icmake/APKBUILD]: When defining custom `prepare` function, always call `default_prepare`, typically as the first command in the function.

File: testing/icmake/APKBUILD [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+	cd "$builddir"
+	sed -i '/^#define DOCDOCDIR/d' INSTALL.im
+	echo '#define DOCDOCDIR "usr/share/doc/icmake"' >> INSTALL.im
+}
+
+build() {
+	cd "$builddir"
+	./icm_prepare /
+	./icm_bootstrap x
```
Comment (jirutka) [inline testing/icmake/APKBUILD]: Just curious, what `x` means here?

File: testing/yodl/APKBUILD [hunk 11]
```diff
@@ -0,0 +1,35 @@
+# Contributor: Shiz <hi@shiz.me>
+# Maintainer: Shiz <hi@shiz.me>
+pkgname=yodl
+pkgver=3.08.02
+pkgrel=0
+pkgdesc="a pre-document language and tools to process it"
```
Comment (jirutka) [inline testing/yodl/APKBUILD]: Please start pkgdesc with a capital letter.

File: testing/yodl/APKBUILD [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+options="!check"
+
+prepare() {
+	cd "$builddir"
+	sed -i 's!DOCDOC\s*=.*$!DOCDOC = BASE + "/share/doc/yodl";!g' INSTALL.im
+}
+
+build() {
+	cd "$builddir"
+	for x in programs macros man ; do
```
Comment (jirutka) [inline testing/yodl/APKBUILD]: Variables used only in a single function should be declared as `local`, e.g. `local x; for x in …; do`. Also it’s better to use a bit more descriptive variable name, e.g. `name` in this case.

File: testing/yodl/APKBUILD [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+pkgrel=0
+pkgdesc="a pre-document language and tools to process it"
+url="https://fbb-git.github.io/yodl/"
+arch="all"
+license="GPL2"
+makedepends="icmake bash perl flex bison"
+subpackages="$pkgname-doc"
+source="yodl-$pkgver.tar.gz::https://github.com/fbb-git/yodl/archive/$pkgver.tar.gz"
+builddir="$srcdir/yodl-$pkgver/yodl"
+options="!check"
```
Comment (jirutka) [inline testing/yodl/APKBUILD]: If upstream does not provide any tests and the project is an application (not a library), then it’s better to add at least very basic check that just calls the produced binary with `--version`. Just to check that the produced binary is not totally broken.

File: testing/yodl/APKBUILD [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+build() {
+	cd "$builddir"
+	for x in programs macros man ; do
+		./build $x
+	done
+}
+
+package() {
+	cd "$builddir"
+	for x in programs macros man ; do
```
Comment (jirutka) [inline testing/yodl/APKBUILD]: Variables used only in a single function should be declared as `local`, e.g. `local x; for x in …; do`. Also it’s better to use a bit more descriptive variable name, e.g. `name` in this case.

File: testing/gpg-remailer/APKBUILD (hunk 4 above)
Comment (Shizmob) [inline testing/gpg-remailer/APKBUILD]: It does not.

File: testing/icmake/APKBUILD (hunk 10 above)
Comment (Shizmob) [inline testing/icmake/APKBUILD]: According to the docs: `(x is a dummy argument required by ./icm_bootstrap to run)`.

Comment (jirutka) [inline testing/icmake/APKBUILD]: LOL, uh, okay. :)

File: testing/bobcat/APKBUILD [hunk 15]
```diff
@@ -0,0 +1,42 @@
+# Contributor: Shiz <hi@shiz.me>
+# Maintainer: Shiz <hi@shiz.me>
+pkgname=bobcat
+pkgver=4.07.00
+pkgrel=0
+pkgdesc="library of C++ classes and templates handling child processes, streams/sockets, shared memory, config files, etc."
+url="https://fbb-git.github.io/bobcat/"
+arch="all"
+license="GPL2+"
```
Comment (jirutka) [inline testing/bobcat/APKBUILD]: Please add `depends_dev="$pkgname=$pkgver-r$pkgrel"`.

**Comment by Shizmob** [comment]
Thanks for the review! I've pushed the needed changes.

File: testing/bobcat/APKBUILD (hunk 15 above)
Comment (Shizmob) [inline testing/bobcat/APKBUILD]: This really oughta be implied in the default dev() implementation...

**Pull request closed without merging**